src/editor/blocks/posts-inserter/utils.js

```diff
@@ -51,6 +51,16 @@ const getSubtitleBlockTemplate = ( post, { subHeadingFontSize, textColor } ) =>
 	return [ 'core/heading', assignFontSize( subHeadingFontSize, attributes ) ];
 };
 
+const getContentBlockTemplate = ( post, { textFontSize, textColor } ) => {
+	let content = post.content.rendered;
+	const contentElement = document.createElement( 'div' );
+	contentElement.innerHTML = content;
+	content = contentElement.textContent || contentElement.innerText || '';
```

Comment:
The use of this strategy can result in a very long unformatted paragraph for posts with larger content. I also experienced a crash while attempting to display a post with custom shortcodes.

Reply:
I do agree with you @miguelpeixe, I would suggest setting the article content (with the word limit that we have currently) only if the post doesn't have an excerpt, but again, would like to have more thoughts on this.

Reply:
An auto-excerpting feature would be nice. If the post does not have an excerpt, the plugin could attempt to take a chunk of the text and use it as excerpt. This would not require any changes in the UI, but work under the hood.

Reply:
Not sure if this is helpful, but I built a function to build [auto-excerpts for Newspack Listings](https://github.com/Automattic/newspack-listings/blob/master/includes/newspack-listings-utils.php#L170-L254) that mimics the core excerpt functionality, but allows certain HTML elements as well. You could maybe adapt this with an absolute max length to avoid bloating the newsletter content.

Reply:
Go to for the auto-excerpting then, @dkoo this will help a lot thank you. I'll adapt this to the already max length option that we have.

Reply:
Also agreed! In that case, we wouldn't have the "content" as a toggle option on the settings, right?

Reply:
Nope, I'll remove the toggle, if the post contains an excerpt it's all good if not I'll generate one from the post content, the excerpt generated can be toggled on/off via the excerpt toggle.

Reply:
I just realized that this is the current behavior 😅; having posts without an excerpt, when added on the post inserter block, the excerpt is automatically generated from the content based on the excerpt max size. Can someone confirm this, please?

Reply:
Yes, looks like it's just what WP does :D Well then, case closed?

Reply:
Agreed! @kariae, feel free to close this PR :)